lib/dotloop/loop.rb

```diff
@@ -2,39 +2,76 @@ module Dotloop
   class Loop
     attr_accessor :client
     BATCH_SIZE = 50
-    # attr_accessor :profile_id
-    # attr_accessor :batch_number
-    # attr_accessor :batch_size
-    # attr_accessor :status_ids
-    # attr_accessor :compliance_status_ids
-    # attr_accessor :tag_ids
-    # attr_accessor :sort_by
-    # attr_accessor :search_query
-    # attr_accessor :tag_names
-    # attr_accessor :created_by_me
 
     def initialize(client:)
       @client = client
     end
 
-    def all(profile_id:)
-      batches = []
+    def all(profile_id:, options: {})
+      loops = []
       batch_number = 1
       loop do
-        batch = retrieve_batch(profile_id: profile_id, batch_size: BATCH_SIZE, batch_number: batch_number)
-        batches += batch
+        options[:batch_size] = BATCH_SIZE
+        options[:batch_number] = batch_number
+        current_loop = batch(profile_id: profile_id, options: options)
+        loops += current_loop
         batch_number += 1
-        break if batch.size < BATCH_SIZE
+        break if current_loop.size < BATCH_SIZE
       end
-      batches
+      loops
     end
 
-    private
-
-    def retrieve_batch(profile_id:, batch_size:, batch_number:)
-      @client.get("/profile/#{profile_id.to_i}/loop", batchSize: batch_size, batchNumber: batch_number).map do |attrs|
+    def batch(profile_id:, options: {})
+      @client.get("/profile/#{profile_id.to_i}/loop", query_params(options)).map do |attrs|
         Dotloop::Models::Loop.new(attrs)
       end
     end
+
+    private
+
+    def query_params(options)
+      {
+        batchNumber:         batch_number(options),
+        batchSize:           batch_size(options),
+        statusIds:           status_ids(options),
+        complianceStatusIds: compliance_status_ids(options),
+        tagIds:              tag_ids(options),
+        sortBy:              options[:sort_by],
+        searchQuery:         options[:search_query],
+        tagNames:            options[:tag_names],
+        createdByMe:         created_by_me(options)
+      }.delete_if { |_, v| v.nil? }
+    end
+
+    def batch_number(options)
+      zero_to_nil(options[:batch_number])
+    end
+
+    def batch_size(options)
+      size = options[:batch_size].to_i
+      return BATCH_SIZE if size < 1 || size > BATCH_SIZE
+      size
+    end
+
+    def status_ids(options)
+      options[:status_ids]&.map { |value| zero_to_nil(value) }
```

Comment:
Why are you nil'ing the zeroes?  Also- after nil'ing shouldn't you #compact the array?